File: src/STech.Core/Domain/Entities/Category.cs
namespace STech.Core.Domain.Entities;

public class Category : BaseEntity
{
    public string Name { get; set; }
    public string? Description { get; set; }
    public int? ParentCategoryID { get; set; }
    public virtual Category ParentCategory { get; set; }
    public int? PictureID { get; set; }
    public virtual Picture Picture { get; set; }
    public virtual List<Product> Products { get; set; }
}

File: src/STech.Client/ClientApp/src/app/shared/models/category.ts
import {IProduct} from "./product";
import {IPicture} from "./picture";

export interface ICategory {
  id: number,
  name: string,
  description: string,
  parentCategoryID: number,
  parentCategory: ICategory,
  pictureID: number,
  picture: IPicture,
  modifiedOn: string,
  products: IProduct[]
}

export interface ICategoryWithChildren {
  category: ICategory,
  children: ICategory[]
}

export interface ICategoryResponse {
  categoryWithChildren: ICategoryWithChildren[]
}
